plugins/network-elements/nuage-vsp/src/com/cloud/network/element/NuageVspElement.java

```diff
@@ -454,6 +563,80 @@ protected boolean applyACLRules(Network network, List<? extends InternalIdentity
     }
 
     @Override
+    public boolean applyNetworkACLs(Network config, List<? extends NetworkACLItem> rules) throws ResourceUnavailableException {
+        if (rules == null || rules.isEmpty()) {
+            s_logger.debug("No rules to apply. So, delete all the existing ACL in VSP from Subnet with uuid " + config.getUuid());
+        } else {
+            s_logger.debug("New rules has to applied. So, delete all the existing ACL in VSP from Subnet with uuid " + config.getUuid());
+        }
+        if (rules != null) {
+            s_logger.debug("Handling applyNetworkACLs for network " + config.getName() + " with " + rules.size() + " Network ACLs");
+            applyACLRules(config, rules, true, null, rules.isEmpty());
+        }
+        return true;
+    }
+
+    @Override
+    public boolean implementVpc(Vpc vpc, DeployDestination dest, ReservationContext context) throws ConcurrentOperationException, ResourceUnavailableException, InsufficientCapacityException {
+        return true;
+    }
+
+    @Override
+    public boolean shutdownVpc(Vpc vpc, ReservationContext context) throws ConcurrentOperationException, ResourceUnavailableException {
+        if (vpc.getState().equals(Vpc.State.Inactive)) {
+            try {
+                Domain vpcDomain = _domainDao.findById(vpc.getDomainId());
+                HostVO nuageVspHost = getNuageVspHost(getPhysicalNetworkId(vpc.getZoneId()));
+                String preConfiguredDomainTemplateName = _configDao.getValue(NuageVspManager.NuageVspVpcDomainTemplateName.key());
+                ShutDownVpcVspCommand cmd = new ShutDownVpcVspCommand(vpcDomain.getUuid(), vpc.getUuid(), preConfiguredDomainTemplateName);
+                ShutDownVpcVspAnswer answer = (ShutDownVpcVspAnswer) _agentMgr.easySend(nuageVspHost.getId(), cmd);
+                if (answer == null || !answer.getResult()) {
+                    s_logger.error("ShutDownVpcVspCommand for VPC " + vpc.getUuid() + " failed");
+                    if ((null != answer) && (null != answer.getDetails())) {
+                        throw new ResourceUnavailableException(answer.getDetails(), Vpc.class, vpc.getId());
+                    }
+                }
+            } catch (Exception e) {
+                s_logger.warn("Failed to destroy VPC in Vsp " + e.getMessage());
```

Comment:
Add the exception to the logging call to provide the stack trace for operational debugging.